lib/src/stylesheet_graph.dart

```diff
@@ -39,7 +40,10 @@ class StylesheetGraph {
     DateTime transitiveModificationTime(StylesheetNode node) {
       return _transitiveModificationTimes.putIfAbsent(node.canonicalUrl, () {
         var latest = node.importer.modificationTime(node.canonicalUrl);
-        for (var upstream in node.upstream.values) {
+        for (var upstream in {
+          ...node.upstream.values,
+          ...node.upstreamImports.values
```

Comment:
I'd use [`Iterable.followedBy()`](https://api.dartlang.org/stable/2.7.0/dart-core/Iterable/followedBy.html) here to avoid copying everything into a set eagerly. There's a possibility we'll visit a file twice, but that seems unlikely and it won't affect the result.

Similar below.